Orange/widgets/model/owneuralnetwork.py

```diff
@@ -0,0 +1,94 @@
+from Orange.widgets import gui
+from Orange.widgets.settings import Setting
+from Orange.widgets.utils.owlearnerwidget import OWBaseLearner
+
+
+class OWNNLearner(OWBaseLearner):
+    name = "Neural Network"
+    description = "A multi-layer perceptron (MLP) algorithm with " \
+                  "backpropagation."
+    icon = "icons/NN.svg"
+    priority = 90
+
+    LEARNER = NNLearner
+
+    activation = ["identity", "logistic", "tanh", "relu"]
+    solver = ["lbfgs", "sgd", "adam"]
+
+    learner_name = Setting("Neural Network")
+    hidden_layers_input = Setting("100")
+    activation_index = Setting(3)
+    solver_index = Setting(2)
+    alpha = Setting(0.0001)
+    max_iterations = Setting(200)
+
+    def add_main_layout(self):
+        box = gui.vBox(self.controlArea, "Network")
+        self.hidden_layers_edit = gui.lineEdit(box, self,
+                                               "hidden_layers_input",
+                                               label="Neurons per hidden "
+                                                     "layer:",
+                                               orientation=Qt.Horizontal,
+                                               callback=self.settings_changed,
+                                               tooltip="A list of integers "
+                                                       "defining neurons. "
+                                                       "Lenght of list "
+                                                       "defines number of "
+                                                       "layers. E.g. 4, 2, 2, 3.")
+        self.activation_combo = gui.comboBox(
+            box, self, "activation_index", orientation=Qt.Horizontal,
+            label="Activation:", items=[i.upper() for i in self.activation],
+            callback=self.settings_changed)
+        self.solver_combo = gui.comboBox(
+            box, self, "solver_index", orientation=Qt.Horizontal,
+            label="Solver:", items=[i.upper() for i in self.solver],
+            callback=self.settings_changed)
+        self.alpha_spin = gui.doubleSpin(
+            box, self, "alpha", 1e-5, 1.0, 1e-2,
+            label="Alpha:", decimals=5, alignment=Qt.AlignRight,
+            callback=self.settings_changed, controlWidth=80)
+        self.max_iter_spin = gui.spin(
+            box, self, "max_iterations", 10, 300, step=10,
+            label="Max iterations:", orientation=Qt.Horizontal,
+            callback=self.settings_changed, controlWidth=80)
+
+    def create_learner(self):
+        return self.LEARNER(
+            hidden_layer_sizes=self.get_hidden_layers(),
+            activation=self.activation[self.activation_index],
+            solver=self.solver[self.solver_index],
+            alpha=self.alpha,
+            max_iter=self.max_iterations,
+            preprocessors=self.preprocessors)
+
+    def get_learner_parameters(self):
+        return (("Hidden layers", self.get_hidden_layers()),
```

Comment:
This prints out as: **Hidden layers:** (100,)

Reply:
Yes. It should. I think.

Reply:
Oh, yes, I see what's the issue.

Reply:
Oh no, I don't.

Reply:
Something like `hidden_layer_sizes=", ".join(map(str, self.get_hidden_layers()))`